### PR TITLE
tests(e2e): extend multi-gw test with scaling to 0

### PR DESCRIPTION
**Pull request by czeslavo**
**What this PR does / why we need it**:

Improves E2E coverage by verifying that when multi-gateway configuration is used, scaling proxy deployment to 0 replicas doesn't break anything and the controller pods remain ready as expected.

Effectively turns the test scenario from "2 replicas, scale to 3 replicas and ensure all configured" into "2 replicas, ensure all configured, scale down to 0, ensure controller pods didn't crash, scale up to 3, ensure all configured".

E2E tests run: https://github.com/Kong/kubernetes-ingress-controller/actions/runs/4317334266

**Which issue this PR fixes**:

Part of #3429.

### CODE REVIEW

File: test/e2e/all_in_one_test.go [hunk 1]
```diff
@@ -336,52 +339,60 @@ func TestDeployAllInOneDBLESS(t *testing.T) {
...
 	verifyIngress(ctx, t, env)
+	ensureAllProxyReplicasAreConfigured(ctx, t, env, deployments.ProxyNN)
 
-	const replicasCount = 3
-	gatewayDeployment := deployments.GetProxy(ctx, t, env)
-	gatewayDeployment.Spec.Replicas = lo.ToPtr(int32(replicasCount))
-	_, err := env.Cluster().Client().AppsV1().Deployments(gatewayDeployment.Namespace).Update(ctx, gatewayDeployment, metav1.UpdateOptions{})
-	require.NoError(t, err)
+	expectedControllerReplicas := *(deployments.GetController(ctx, t, env).Spec.Replicas)
+	t.Logf("store expected controller replicas count = %d", expectedControllerReplicas)
```
Comment (randmonkey) [inline test/e2e/all_in_one_test.go:345]: Is it necessary to store the expected controller replicas? Did you intend to scale the controller deployment here?

File: test/e2e/all_in_one_test.go [hunk 2]
```diff
@@ -336,52 +339,60 @@ func TestDeployAllInOneDBLESS(t *testing.T) {
...
+	scaleDeployment(ctx, t, env, deployments.ProxyNN, 3)
+	ensureAllProxyReplicasAreConfigured(ctx, t, env, deployments.ProxyNN)
+}
 
-	t.Log("confirming that all dataplanes got the config")
+func ensureAllProxyReplicasAreConfigured(ctx context.Context, t *testing.T, env environments.Environment, proxyDeploymentNN types.NamespacedName) {
+	forDeployment := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("app=%s", proxyDeploymentNN.Name),
+	}
+	podList, err := env.Cluster().Client().CoreV1().Pods(proxyDeploymentNN.Namespace).List(ctx, forDeployment)
```
Comment (randmonkey) [inline test/e2e/all_in_one_test.go:365]: I think we could use `getPodByLabels` to do this.

**Review by randmonkey — COMMENTED** [review]
Generally LGTM, only some minor comments remaining.

File: test/e2e/all_in_one_test.go (hunk 1 above)
Comment (czeslavo) [reply test/e2e/all_in_one_test.go]: Good catch, yeah, it's not used. I meant to use it to ensure that after scaling the proxy down to 0, the count of the controller's ready pods doesn't change. Fixed that: d66c33d

File: test/e2e/all_in_one_test.go (hunk 2 above)
Comment (czeslavo) [reply test/e2e/all_in_one_test.go]: Yeah, good point, let's reuse the goodies we already got in the tool belt :) d66c33d

**Review by pmalek — APPROVED** [review]

**Pull request merged**